x/oracle/keeper/history_test.go

```diff
@@ -154,3 +154,231 @@ func TestRemoveHistoryEntryBeforeTime(t *testing.T) {
+	require.NoError(t, err)
+	require.Equal(t, phEntrysDen[0], phStoreDen)
+	phStoreDen, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Den", phEntrysDen[0].PriceUpdateTime.Add(time.Minute))
+	require.NoError(t, err)
+	require.Equal(t, phEntrysDen[0], phStoreDen)
+	phStoreDen, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Den", phEntrysDen[1].PriceUpdateTime)
+	require.NoError(t, err)
+	require.Equal(t, phEntrysDen[1], phStoreDen)
+	phStoreDen, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Den", phEntrysDen[1].PriceUpdateTime.Add(time.Minute))
+	require.NoError(t, err)
+	require.Equal(t, phEntrysDen[1], phStoreDen)
+
+	phStoresDen, err := oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Den",
+		phEntrysDen[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrysDen[2].PriceUpdateTime.Add(time.Minute),
+	)
+	require.NoError(t, err)
+	require.Equal(t, phStoresDen, phEntrysDen)
+
+	phStoresDen, err = oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Den",
+		phEntrysDen[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrysDen[1].PriceUpdateTime,
+	)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(phStoresDen))
+	require.Equal(t, phStoresDen[0], phEntrysDen[0])
+	require.Equal(t, phStoresDen[1], phEntrysDen[1])
+
+	// checks for token with denom: "Juno"
+	phStoreJuno, err := oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[0].PriceUpdateTime.Add(-time.Minute))
+	require.Error(t, err)
+	require.Equal(t, types.PriceHistoryEntry{}, phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[0].PriceUpdateTime)
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[0], phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[0].PriceUpdateTime.Add(time.Minute))
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[0], phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[1].PriceUpdateTime)
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[1], phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[1].PriceUpdateTime.Add(time.Minute))
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[1], phStoreJuno)
+
+	phStoresdjuno, err := oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Juno",
+		phEntrysJuno[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrysJuno[2].PriceUpdateTime.Add(time.Minute),
+	)
+	require.NoError(t, err)
+	require.Equal(t, phStoresdjuno, phEntrysJuno)
+
+	phStoresdjuno, err = oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Juno",
+		phEntrysJuno[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrysJuno[1].PriceUpdateTime,
+	)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(phStoresdjuno))
+	require.Equal(t, phStoresdjuno[0], phEntrysJuno[0])
+	require.Equal(t, phStoresdjuno[1], phEntrysJuno[1])
+}
+
+func TestStoreAndGetNullHistoricalData(t *testing.T) {
+	timeNow := time.Now().UTC()
```

Comment:
Using a driven table test for this case could make more clear && easier to debug!